frontend-react/src/hooks/UseOktaMemberships.ts

```diff
@@ -0,0 +1,151 @@
+import React, { useEffect, useReducer } from "react";
+import { AccessToken } from "@okta/okta-auth-js";
+
+import { getOktaGroups, parseOrgName } from "../utils/OrganizationUtils";
+
+export enum MemberType {
+    SENDER = "sender",
+    RECEIVER = "receiver",
+    PRIME_ADMIN = "prime-admin",
+    NON_STAND = "non-standard",
+}
+
+export enum MembershipActionType {
+    SWITCH = "switch",
+    UPDATE = "update",
+    ADMIN_OVERRIDE = "override",
+}
+
+interface MembershipSettings {
+    // The org header value
+    parsedName: string;
+    // The type of membership
+    memberType: MemberType;
+}
+
+export interface MembershipState {
+    active?: MembershipSettings;
```

Comment:
It's kinda like a private method I guess...like, I can assign a value to `active` and TS _will_ check it against that type, but I don't have access to the type shape to assign to a variable to get typing help. But a good idea is typically to export all of them. I think I didn't need to with this because the only place handling _creating_ settings is in this module